usr/share/rear/restore/YUM/default/940_generate_fstab.sh

```diff
@@ -29,7 +29,7 @@ while read keyword device_node junk ; do
     # Do not use a possibly outdated UUID from LAYOUT_FILE but determine the actual one in the current system:
     uuid=$( for uuid in * ; do readlink -e $uuid | grep -q $device_node && echo $uuid || true ; done  )
     # If fstab already contains an entry for this swap device, do nothing
-    if $(egrep -q "^UUID=$uuid\s+swap" $TARGET_FS_ROOT/etc/fstab) -o $(egrep -q "^$device_node\s+swap" $TARGET_FS_ROOT/etc/fstab) ; then
+    if $(egrep -q "^UUID=$uuid\s+swap" $TARGET_FS_ROOT/etc/fstab) || $(egrep -q "^$device_node\s+swap" $TARGET_FS_ROOT/etc/fstab) ; then
```

Comment:
I think in general a command like
<pre>
if $( COMMAND1 ) || $( COMMAND2 ) ; then COMMAND3 ; fi
</pre>
does not work as expected e.g. because:
<pre>
# if $( cat qqq ) || $( echo hello ) ; then echo Y ; else echo N ; fi
cat: qqq: No such file or directory
If 'hello' is not a typo you can use command-not-found to lookup the package that contains it
N
</pre>
I guess what was actually meant is
<pre>
if COMMAND1 || COMMAND2 ; then COMMAND3 ; fi
</pre>

Reply:
Thanks!  I'm committing an update for this right now.